boost_predef/os/haiku.h

```diff
@@ -0,0 +1,46 @@
+/*
```

Comment:
apparently you dumped in the whole boost_predef directory in the repository. this means we also have to manage updates somehow which could be a problem. I suggest using a submodule or subtree for this since this can become quite a lot of clutter.

Reply:
No; I only added os.h and its explicit dependencies to the tree.
As for updates, perhaps but in reality I doubt it will be tremendously important (and in fact the version I added isn't even the latest boost because I didn't have it on hand). I don't think we need to worry too much about this because,

1) It's just a bunch of headers with mostly macro definitions.
2) It's only to be used on platforms that rely on obsolete versions of boost (<= 1.54), think of it as a polyfill.